gcloud-java-dns/src/test/java/com/google/gcloud/dns/ITDnsTest.java

```diff
@@ -0,0 +1,1268 @@
+    ImmutableList<ChangeRequest> descending = ImmutableList.copyOf(DNS.listChangeRequests(
+        ZONE1.name(),
+        Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.DESCENDING)).iterateAll());
+    int size = 5;
+    assertEquals(size, descending.size());
+    assertEquals(size, ascending.size());
+    for (int i = 0; i < size; i++) {
+      assertEquals(descending.get(i), ascending.get(size - i - 1));
+    }
+    // field options
+    changes = ImmutableList.copyOf(DNS.listChangeRequests(ZONE1.name(),
+        Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+        Dns.ChangeRequestListOption.fields(Dns.ChangeRequestField.ADDITIONS)).iterateAll());
+    change = changes.get(1);
+    assertEquals(CHANGE_ADD_ZONE1.additions(), change.additions());
+    assertTrue(change.deletions().isEmpty());
+    assertEquals("1", change.id());
+    assertNull(change.startTimeMillis());
+    assertNull(change.status());
+    changes = ImmutableList.copyOf(DNS.listChangeRequests(ZONE1.name(),
+        Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+        Dns.ChangeRequestListOption.fields(Dns.ChangeRequestField.DELETIONS)).iterateAll());
+    change = changes.get(2);
+    assertTrue(change.additions().isEmpty());
+    assertNotNull(change.deletions());
+    assertEquals("2", change.id());
+    assertNull(change.startTimeMillis());
+    assertNull(change.status());
+    changes = ImmutableList.copyOf(DNS.listChangeRequests(ZONE1.name(),
+        Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+        Dns.ChangeRequestListOption.fields(Dns.ChangeRequestField.ID)).iterateAll());
+    change = changes.get(1);
+    assertTrue(change.additions().isEmpty());
+    assertTrue(change.deletions().isEmpty());
+    assertEquals("1", change.id());
+    assertNull(change.startTimeMillis());
+    assertNull(change.status());
+    changes = ImmutableList.copyOf(DNS.listChangeRequests(ZONE1.name(),
+        Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+        Dns.ChangeRequestListOption.fields(Dns.ChangeRequestField.START_TIME)).iterateAll());
+    change = changes.get(1);
+    assertTrue(change.additions().isEmpty());
+    assertTrue(change.deletions().isEmpty());
+    assertEquals("1", change.id());
+    assertNotNull(change.startTimeMillis());
+    assertNull(change.status());
+    changes = ImmutableList.copyOf(DNS.listChangeRequests(ZONE1.name(),
+        Dns.ChangeRequestListOption.sortOrder(Dns.SortingOrder.ASCENDING),
+        Dns.ChangeRequestListOption.fields(Dns.ChangeRequestField.STATUS)).iterateAll());
+    change = changes.get(1);
+    assertTrue(change.additions().isEmpty());
+    assertTrue(change.deletions().isEmpty());
+    assertEquals("1", change.id());
+    assertNull(change.startTimeMillis());
+    assertEquals(ChangeRequest.Status.DONE, change.status());
+  }
+
+  @Test
+  public void testListChangesUsingZoneObject() {
```

Comment:
Nit: shouldn't be needed as we unit test Zone and have integration test for the service